Code/Common/src/sitkPimpleImageBase.hxx

```diff
@@ -722,6 +724,38 @@ namespace itk
                             << "!" );
       }
 
+    template< typename TPixelType, unsigned int NDimension >
+    bool
+    InternalCheckBuffer(const itk::Image<TPixelType, NDimension> *image)
+      {
+        auto container = image->GetPixelContainer();
+        if (container->GetReferenceCount() != 1)
+          {
+          sitkExceptionMacro( << "The image pixel container is shared by other resources and presents aliasing issue.");
+          }
+        return true;
+      }
+
+    template< typename TPixelType, unsigned int NDimension >
+    bool
+    InternalCheckBuffer(const itk::VectorImage<TPixelType, NDimension> *image)
+      {
+        auto container = image->GetPixelContainer();
+        if (container->GetReferenceCount() != 1)
+          {
+          sitkExceptionMacro( << "The image pixel container is shared by other resources and presents aliasing issues.");
```

Comment:
Maybe modify the exception message between the scalar image and vector image, just to make it easier to localize issues?

Reply:
The exceptions contain the line number for localization, but we can certainly add "vector image" here.